server/src/main/java/com/linecorp/centraldogma/server/internal/admin/auth/AbstractAuthorizer.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2025 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.centraldogma.server.internal.admin.auth;
+
+import java.util.concurrent.CompletionStage;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.auth.BasicToken;
+import com.linecorp.armeria.common.auth.OAuth2Token;
+import com.linecorp.armeria.common.util.UnmodifiableFuture;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.auth.AuthTokenExtractors;
+import com.linecorp.armeria.server.auth.Authorizer;
+
+import io.netty.util.AttributeKey;
+
+public abstract class AbstractAuthorizer implements Authorizer<HttpRequest> {
+
+    private static final AttributeKey<String> TOKEN_KEY =
+            AttributeKey.valueOf(AbstractAuthorizer.class, "TOKEN_KEY");
+
+    private static final String BASIC_USERNAME = "dogma";
+
+    @Override
+    public CompletionStage<Boolean> authorize(ServiceRequestContext ctx, HttpRequest req) {
+        final String token = extractToken(ctx, req);
+        if (token == null) {
+            return UnmodifiableFuture.completedFuture(false);
+        }
+
+        return authorize(ctx, req, token);
+    }
+
+    protected abstract CompletionStage<Boolean> authorize(ServiceRequestContext ctx, HttpRequest req,
+                                                          String accessToken);
+
+    @Nullable
+    private static String extractToken(ServiceRequestContext ctx, HttpRequest req) {
+        String token = ctx.attr(TOKEN_KEY);
```

Comment:
nit; personally I'm not sure if this additional complexity is worth it compared to the cost of header lookups, but I understood the changes.

Reply:
A parsed token is stored in the attributes to avoid duplicate parsing logic, not just to skip the headers lookup.